code/js/modules/BaseController.js

```diff
@@ -210,11 +216,23 @@
    * @return {{song: {String}, artist: {String}, isPlaying: {Boolean}, siteName: {String}}}
    */
   BaseController.prototype.getStateData = function() {
+    var vCanDislike = !!(this.selectors.dislike && this.doc().querySelector(this.selectors.dislike)),
+        vCanPlayPrev = !!(this.selectors.playPrev && this.doc().querySelector(this.selectors.playPrev)),
+        vCanPlayPause = !!((this.selectors.playPause && this.doc().querySelector(this.selectors.playPause)) || (this.selectors.play && this.doc().querySelector(this.selectors.play)) || (this.selectors.pause && this.doc().querySelector(this.selectors.pause))),
```

Comment:
Would you agree then that `hasPlayer` serves no purpose, as long as we know `canPlayPause`  is true then `hasPlayer` would always be true?

If you agree then I can reduce this whole code block by directly setting the return values instead of declaring variables first. The variables were only used because `hasPlayer` depended on the others.

I will remove the `playPause` block from popup.js and the purpose of the variable existing here will take the place of `hasPlayer` which is meant to help us determine if a player's control container should be hidden from the popup if a tab doesn't actually have a player (ex:amazon)

Let me know what you think, I might just make the commit and you can review it by seeing it.